src/ATen/function_wrapper.py

```diff
@@ -75,6 +75,22 @@
     return static_cast<Type*>(this)->${method_prefix}${api_name}(${sparse_actuals});
 }""")
 
+WRAP_DIM_GEN = CodeTemplate("""\
+auto ${target}${dim}_ = std::max<${type}>(${target_dim_expr}, 0);
+if (${target}${dim}_ <= 0) {
+  std::ostringstream oss;
+  oss << "dimension specified as " << ${dim} << " but tensor has no dimensions";
```

Comment:
I would modify the error message to indicate that the failure occurred because `${target}` is a Scalar, or is undefined, i.e.

`Dimension specified as 0 but tensor is [a Scalar|undefined]` or what have you. I trust in your use of the English language to be better than mine.

Reply:
nit:  we should not mention "Scalar" in the public API. It is an implementation detail of ATen's internals only.